### ISSUE TITLE
https://www.terraform.io/docs/providers/oci/r/database_autonomous_database.html issues with the data_safe

### ISSUE THREAD

**Issue by eugsim1** [terraform-providers/terraform-provider-oci#943]
Hello the tag data_safe_status in  resource "oci_database_autonomous_database" "autonomous_data_warehouse" { is not working

  count               = var.instance_count
  admin_password           = "${var.db_admin_password}"
  compartment_id           = "${var.compartment_ocid}"
  cpu_core_count           = "1"
  data_storage_size_in_tbs = "1"
  db_name                  = "ebsadw${count.index}"
  #data_safe_status         = "REGISTERED" 

same for the data part of the call https://www.terraform.io/docs/providers/oci/d/database_autonomous_database.html

the oci sdk go give the sate correctly am i doing someting wrong ?
output of the go sdk :
Autonomous Db[0]: ebsadw5 1 1 TERMINATED student5 18c NOT_REGISTERED
Autonomous Db[1]: ebsadw2 1 1 TERMINATED student2 18c NOT_REGISTERED
Autonomous Db[2]: ebsadw3 1 1 TERMINATED student3 18c FAILED
Autonomous Db[3]: ebsadw6 1 1 TERMINATED student6 18c NOT_REGISTERED
Autonomous Db[4]: ebsadw0 1 1 AVAILABLE student0 18c NOT_REGISTERED
Autonomous Db[5]: ebsadw4 1 1 TERMINATED student4 18c FAILED
Autonomous Db[6]: ebsadw1 1 1 AVAILABLE student1 18c NOT_REGISTERED
Autonomous Db[7]: ebsadw7 1 1 TERMINATED student7 18c REGISTERED

data "oci_database_autonomous_databases" "autonomous_data_warehouses" {
  count               = var.instance_count
  compartment_id = "${var.compartment_ocid}"

  #Optional
  display_name = "${oci_database_autonomous_database.autonomous_data_warehouse[count.index].display_name}"
  db_workload  = "${var.autonomous_data_warehouse_db_workload}"
}

output "autonomous_data_warehouse_data_safe_status" {
  value = "${oci_database_autonomous_database.autonomous_data_warehouse.* }"
}
autonomous_data_warehouse_data_safe_status = [
  {
    "admin_password" = "BEstrO0ng_#12"
    "compartment_id" = "ocid1.compartment.oc1..aaaaaaaaec5to2hjrxvzpsybpo776hs25ybtqptkksjzxn4khlnqsvxjogpa"
    "connection_strings" = [
      {
        "all_connection_strings" = {
          "HIGH" = "adb.eu-frankfurt-1.oraclecloud.com:1522/vx7tqpyaop2tflx_ebsadw0_high.adwc.oraclecloud.com"
          "LOW" = "adb.eu-frankfurt-1.oraclecloud.com:1522/vx7tqpyaop2tflx_ebsadw0_low.adwc.oraclecloud.com"
          "MEDIUM" = "adb.eu-frankfurt-1.oraclecloud.com:1522/vx7tqpyaop2tflx_ebsadw0_medium.adwc.oraclecloud.com"
        }
        "dedicated" = ""
        "high" = "adb.eu-frankfurt-1.oraclecloud.com:1522/vx7tqpyaop2tflx_ebsadw0_high.adwc.oraclecloud.com"
        "low" = "adb.eu-frankfurt-1.oraclecloud.com:1522/vx7tqpyaop2tflx_ebsadw0_low.adwc.oraclecloud.com"
        "medium" = "adb.eu-frankfurt-1.oraclecloud.com:1522/vx7tqpyaop2tflx_ebsadw0_medium.adwc.oraclecloud.com"
      },
    ]
    "connection_urls" = [
      {
        "apex_url" = "https://VX7TQPYAOP2TFLX-EBSADW0.adb.eu-frankfurt-1.oraclecloudapps.com/ords/apex"
        "sql_dev_web_url" = "https://VX7TQPYAOP2TFLX-EBSADW0.adb.eu-frankfurt-1.oraclecloudapps.com/ords/admin/_sdw/?nav=worksheet"
      },
    ]
    "cpu_core_count" = 1
    "data_storage_size_in_tbs" = 1
    "db_name" = "ebsadw0"
    "db_version" = "18c"
    "db_workload" = "DW"
    "defined_tags" = {}
    "display_name" = "student0"
    "freeform_tags" = {}
    "id" = "ocid1.autonomousdatabase.oc1.eu-frankfurt-1.abtheljsyp5dgeunkf7buqsf2plvgjvuymuzmssve2ucll7diobf4cwihxnq"
    "is_auto_scaling_enabled" = false
    "is_dedicated" = false
    "is_free_tier" = false
    "is_preview" = false
    "license_model" = "BRING_YOUR_OWN_LICENSE"
    "service_console_url" = "https://adb.eu-frankfurt-1.oraclecloud.com/console/index.html?tenant_name=OCID1.TENANCY.OC1..AAAAAAAAHE24EABQYIYLME5ZI2GPERNBX7BLO5VS3HKNTXVLFVTZV2CIIX5Q&database_name=EBSADW0&service_type=ADW"
    "state" = "AVAILABLE"
    "system_tags" = {}
    "time_created" = "2019-12-17 18:34:08.612 +0000 UTC"
  },
  {
    "admin_password" = "BEstrO0ng_#12"
    "compartment_id" = "ocid1.compartment.oc1..aaaaaaaaec5to2hjrxvzpsybpo776hs25ybtqptkksjzxn4khlnqsvxjogpa"
    "connection_strings" = [
      {
        "all_connection_strings" = {
          "HIGH" = "adb.eu-frankfurt-1.oraclecloud.com:1522/vx7tqpyaop2tflx_ebsadw1_high.adwc.oraclecloud.com"
          "LOW" = "adb.eu-frankfurt-1.oraclecloud.com:1522/vx7tqpyaop2tflx_ebsadw1_low.adwc.oraclecloud.com"
          "MEDIUM" = "adb.eu-frankfurt-1.oraclecloud.com:1522/vx7tqpyaop2tflx_ebsadw1_medium.adwc.oraclecloud.com"
        }
        "dedicated" = ""
        "high" = "adb.eu-frankfurt-1.oraclecloud.com:1522/vx7tqpyaop2tflx_ebsadw1_high.adwc.oraclecloud.com"
        "low" = "adb.eu-frankfurt-1.oraclecloud.com:1522/vx7tqpyaop2tflx_ebsadw1_low.adwc.oraclecloud.com"
        "medium" = "adb.eu-frankfurt-1.oraclecloud.com:1522/vx7tqpyaop2tflx_ebsadw1_medium.adwc.oraclecloud.com"
      },
    ]
    "connection_urls" = [
      {
        "apex_url" = "https://VX7TQPYAOP2TFLX-EBSADW1.adb.eu-frankfurt-1.oraclecloudapps.com/ords/apex"
        "sql_dev_web_url" = "https://VX7TQPYAOP2TFLX-EBSADW1.adb.eu-frankfurt-1.oraclecloudapps.com/ords/admin/_sdw/?nav=worksheet"
      },
    ]
    "cpu_core_count" = 1
    "data_storage_size_in_tbs" = 1
    "db_name" = "ebsadw1"
    "db_version" = "18c"
    "db_workload" = "DW"
    "defined_tags" = {}
    "display_name" = "student1"
    "freeform_tags" = {}
    "id" = "ocid1.autonomousdatabase.oc1.eu-frankfurt-1.abtheljrql2uy2aiaygnbp2pml5vzyhlqx7hupzdxz4cie2r5lato46fygwa"
    "is_auto_scaling_enabled" = false
    "is_dedicated" = false
    "is_free_tier" = false
    "is_preview" = false
    "license_model" = "BRING_YOUR_OWN_LICENSE"
    "service_console_url" = "https://adb.eu-frankfurt-1.oraclecloud.com/console/index.html?tenant_name=OCID1.TENANCY.OC1..AAAAAAAAHE24EABQYIYLME5ZI2GPERNBX7BLO5VS3HKNTXVLFVTZV2CIIX5Q&database_name=EBSADW1&service_type=ADW"
    "state" = "AVAILABLE"
    "system_tags" = {}
    "time_created" = "2019-12-17 18:32:11.563 +0000 UTC"
  },
]

**Comment by abhilash-av** [comment]
@eugsim1 Could you please share the full tettaform config and debug logs
Also i would like to inform you that 
`Support for data safe integration in oci_database_autonomous_database resource` is available from provider version 3.52.0. Please confirm you that you are using the provider version >= 3.52.0 and let me know if it works

**Comment by eugsim1** [comment]
Hello i have removed the .terraform and recreated with terraform init now i see that im getting an upgraded provider * provider.local: version = "~> 1.4"
* provider.oci: version = "~> 3.55" and the tags with data_safe seems to pass but the data_safe fails, i will downgrade to oci.version 3.52 as you suggested

**Comment by abhilash-av** [comment]
@eugsim1 `Support for data safe integration in oci_database_autonomous_database resource` was available since provider version 3.52.0, using any version greater than 3.52.0 should still support this feature. There is no need to downgrade. Also, I tried the https://github.com/terraform-providers/terraform-provider-oci/blob/master/examples/database/atp-d/main.tf with data_safe_status         = "REGISTERED" and the example worked fine. Closing this issue for now.
